pkg/test/framework/components/istio/configmap.go

```diff
@@ -299,6 +300,11 @@ func (mc *meshConfig) UpdateMeshConfig(t resource.Context, update func(*meshconf
 
 			cfgMap, err := mc.getConfigMap(c, cfgMapName)
 			if err != nil {
+				// Remote clusters typically don't have mesh config, allow it to skip
+				if c.IsRemote() && kerrors.IsNotFound(err) {
```

Comment:
Why not guard the call to `getConfigMap` with `!c.IsRemote()`? Do we have a use case where we do want to load it if it's there?

Reply:
Actually, the guard would need to be c.IsConfig() - i.e., a primary or remote config cluster should have the configMap, but pure remote should not.

Reply:
Well we used to deploy it, so figured might as well patch it if its there? I don't think it matters for our test though

Reply:
Yeah, agree, tests are fine either way.

Reply:
Actually, I think I just remembered why we used to have it. For quite some time remote clusters (Nate's initial impl) weren't truly remote. The remote istiod was just used for discovery, while a local istiod was still deployed for CA. Anyway, it shouldn't be there any more but harmless if we handle it if it is.